{
    "comments": [
        {
            "author": null,
            "body": "output of catalina.out, when you deploy an application, where a context exists\nin server.xml, but $CATALINA_BASE/webapps is empty.\n\nINFO: HTMLManager: start: Starting web application at '/yyy'\n12.10.2007 13:54:56 org.apache.catalina.core.StandardContext resourcesStart\nSCHWERWIEGEND: Error starting static Resources\njava.lang.IllegalArgumentException: Document base\n/home/tomcat/webapps/yyy/webapps/yyy does not exist or is not a readable directory\n        at\norg.apache.naming.resources.FileDirContext.setDocBase(FileDirContext.java:141)\n        at\norg.apache.catalina.core.StandardContext.resourcesStart(StandardContext.java:3855)\n        at org.apache.catalina.core.StandardContext.start(StandardContext.java:4024)\n        at\norg.apache.catalina.manager.ManagerServlet.start(ManagerServlet.java:1175)\n        at\norg.apache.catalina.manager.HTMLManagerServlet.start(HTMLManagerServlet.java:527)\n        at\norg.apache.catalina.manager.HTMLManagerServlet.doGet(HTMLManagerServlet.java:104)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java:690)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java:803)\n        at\norg.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:269)\n        at\norg.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:188)\n        at\norg.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:213)\n        at\norg.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:174)\n        at\norg.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:525)\n        at\norg.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)\n        at\norg.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:117)\n        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:544)\n        at\norg.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:108)\n        at\norg.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:151)\n        at\norg.apache.coyote.http11.Http11Processor.process(Http11Processor.java:874)\n        at\norg.apache.coyote.http11.Http11BaseProtocol$Http11ConnectionHandler.processConnection(Http11BaseProtocol.java:665)\n        at\norg.apache.tomcat.util.net.PoolTcpEndpoint.processSocket(PoolTcpEndpoint.java:528)\n        at\norg.apache.tomcat.util.net.LeaderFollowerWorkerThread.runIt(LeaderFollowerWorkerThread.java:81)\n        at\norg.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:689)\n        at java.lang.Thread.run(Thread.java:595)\n12.10.2007 13:54:56 org.apache.catalina.core.StandardContext start",
            "date": "20071012T05:24:40",
            "id": 0
        },
        {
            "author": null,
            "body": "This is fixed in trunk - by providing an error message if you try and upload a\nfile when the associated context is defined in server.xml\n\nI have proposed the fix for back-porting to 6.0.x and 5.5.x",
            "date": "20071201T10:32:18",
            "id": 1
        },
        {
            "author": null,
            "body": "Fixed in 5.5.x and will be included in 5.5.26 onwards.",
            "date": "20080106T15:13:41",
            "id": 2
        }
    ],
    "component": "Connector:HTTP",
    "description": "if a context for the war file exists in server.xml, the manager uploads the new\n.war file, but does not expand it, if the necessary directory does not exist,\nand if the directory exists, it is not overwritten, so you see the old application.\n\nsee server.xml\n\n<Host name=\"xxx\" appBase=\"webapps\" unpackWARs=\"true\">\n        <Context path=\"/yyy\">\n...\n\nif I don't have a context for yyy in server.xml, autodeploy works as expected.\n\nEven if you say, remove context from the manager application, it is always there\nin the application view, you can't remove it.\n\nAnd finally, it never cleans up the work directory for the application yyy,\nwhich I guess, is just a subsequent error of the above behaviour.",
    "hasPatch": false,
    "hasScreenshot": false,
    "id": "43611",
    "issuetypeClassified": "BUG",
    "issuetypeTracker": "BUG",
    "priority": "P2 major",
    "product": "Tomcat 5",
    "project": "TOMCAT",
    "summary": "autodeployment with unpackWARs=\"true\" does not work, when a context for that war file exists in server.xml",
    "systemSpecification": true,
    "version": "5.5.24"
}